### ISSUE TITLE
Set() with non-primitive values is deprecated (node) and will stop working in the next major release. Warning

### ISSUE THREAD

**Issue by LeeBergstrand** [tickbin/tickbin#190]
I'm getting this warning when call tick.

```
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x2b348cac9fa9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x38c0f5e07944] (this=0x137c153d8bd9 <an Object with map 0x38824ac17e49>#1#,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    2: load [module.js:456] [pc=0x38c0f5c39152] (this=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:415] [pc=0x38c0f5c38c7d] (this=0x2b348ca04189 <undefined>,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    4: _load [module.js:~381] [pc=0x38c0f5db9bff] (this=0x137c153d8c59 <JS Func 1: v8::Template::Set(v8::Local<v8::Name>, v8::Local<v8::Data>, v8::PropertyAttribute)
 2: leveldown::Database::Init()
 3: leveldown::Init(v8::Local<v8::Object>)
 4: node::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)
 5: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&))
 6: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>)
 7: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*)
 8: 0x38c0f5b0961b
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x2b348cac9fa9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x38c0f5e07944] (this=0x137c153d8bd9 <an Object with map 0x38824ac17e49>#1#,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    2: load [module.js:456] [pc=0x38c0f5c39152] (this=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:415] [pc=0x38c0f5c38c7d] (this=0x2b348ca04189 <undefined>,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    4: _load [module.js:~381] [pc=0x38c0f5db9bff] (this=0x137c153d8c59 <JS Func
==== C stack trace ===============================

 1: v8::Template::Set(v8::Local<v8::Name>, v8::Local<v8::Data>, v8::PropertyAttribute)
 2: leveldown::Database::Init()
 3: leveldown::Init(v8::Local<v8::Object>)
 4: node::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)
 5: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&))
 6: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>)
 7: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*)
 8: 0x38c0f5b0961b
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x2b348cac9fa9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x38c0f5e07944] (this=0x137c153d8bd9 <an Object with map 0x38824ac17e49>#1#,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    2: load [module.js:456] [pc=0x38c0f5c39152] (this=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:415] [pc=0x38c0f5c38c7d] (this=0x2b348ca04189 <undefined>,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    4: _load [module.js:~381] [pc=0x38c0f5db9bff] (this=0x137c153d8c59 <JS Func
==== C stack trace ===============================

 1: v8::Template::Set(v8::Local<v8::Name>, v8::Local<v8::Data>, v8::PropertyAttribute)
 2: leveldown::Database::Init()
 3: leveldown::Init(v8::Local<v8::Object>)
 4: node::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)
 5: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&))
 6: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>)
 7: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*)
 8: 0x38c0f5b0961b
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x2b348cac9fa9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x38c0f5e07944] (this=0x137c153d8bd9 <an Object with map 0x38824ac17e49>#1#,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    2: load [module.js:456] [pc=0x38c0f5c39152] (this=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:415] [pc=0x38c0f5c38c7d] (this=0x2b348ca04189 <undefined>,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    4: _load [module.js:~381] [pc=0x38c0f5db9bff] (this=0x137c153d8c59 <JS Func 1: v8::Template::Set(v8::Local<v8::Name>, v8::Local<v8::Data>, v8::PropertyAttribute)
 2: leveldown::Database::Init()
 3: leveldown::Init(v8::Local<v8::Object>)
 4: node::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)
 5: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&))
 6: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>)
 7: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*)
 8: 0x38c0f5b0961b
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x2b348cac9fa9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x38c0f5e07944] (this=0x137c153d8bd9 <an Object with map 0x38824ac17e49>#1#,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    2: load [module.js:456] [pc=0x38c0f5c39152] (this=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:415] [pc=0x38c0f5c38c7d] (this=0x2b348ca04189 <undefined>,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    4: _load [module.js:~381] [pc=0x38c0f5db9bff] (this=0x137c153d8c59 <JS Func 1: v8::Template::Set(v8::Local<v8::Name>, v8::Local<v8::Data>, v8::PropertyAttribute)
 2: leveldown::Database::Init()
 3: leveldown::Init(v8::Local<v8::Object>)
 4: node::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)
 5: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&))
 6: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>)
 7: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*)
 8: 0x38c0f5b0961b
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x2b348cac9fa9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x38c0f5e07944] (this=0x137c153d8bd9 <an Object with map 0x38824ac17e49>#1#,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    2: load [module.js:456] [pc=0x38c0f5c39152] (this=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:415] [pc=0x38c0f5c38c7d] (this=0x2b348ca04189 <undefined>,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    4: _load [module.js:~381] [pc=0x38c0f5db9bff] (this=0x137c153d8c59 <JS Func 1: v8::Template::Set(v8::Local<v8::Name>, v8::Local<v8::Data>, v8::PropertyAttribute)
 2: leveldown::Database::Init()
 3: leveldown::Init(v8::Local<v8::Object>)
 4: node::DLOpen(v8::FunctionCallbackInfo<v8::Value> const&)
 5: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&))
 6: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>)
 7: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*)
 8: 0x38c0f5b0961b
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x2b348cac9fa9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x38c0f5e07944] (this=0x137c153d8bd9 <an Object with map 0x38824ac17e49>#1#,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    2: load [module.js:456] [pc=0x38c0f5c39152] (this=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    3: tryModuleLoad(aka tryModuleLoad) [module.js:415] [pc=0x38c0f5c38c7d] (this=0x2b348ca04189 <undefined>,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    4: _load [module.js:~381] [pc=0x38c0f5db9bff] (this=0x137c153d8c59 <JS Func 5: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&))
 6: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::(anonymous namespace)::BuiltinArguments<(v8::internal::BuiltinExtraArguments)1>)
 7: v8::internal::Builtin_HandleApiCall(int, v8::internal::Object**, v8::internal::Isolate*)
 8: 0x38c0f5b0961b
(node) v8::ObjectTemplate::Set() with non-primitive values is deprecated
(node) and will stop working in the next major release.

==== JS stack trace =========================================

Security context: 0x2b348cac9fa9 <JS Object>#0#
    1: .node [module.js:568] [pc=0x38c0f5e07944] (this=0x137c153d8bd9 <an Object with map 0x38824ac17e49>#1#,module=0x3ad019583721 <a Module with map 0x38824ac92291>#2#,filename=0x3ad0195836f9 <String[87]: /usr/local/lib/node_modules/tickbin/node_modules/leveldown/build/Release/leveldown.node>)
    2: load [module.js:45Usage: tick <command> [options]
```

**Comment by chrisfosterelli** [comment]
Oh, interesting... I don't think we've seen this. What node version are you
running? (node -v)

**Comment by LeeBergstrand** [comment]
@chrisfosterelli v6.1.0

**Comment by chrisfosterelli** [comment]
Thanks, will look into it!

**Comment by chrisfosterelli** [comment]
Found the current cause: PouchDB is using an older version of leveldown, which uses an older version of a package called nan. The nan package triggers the error. This has been fixed upstream, but PouchDB pins version numbers and has not released a new version with the updated leveldown.

They have a [pull request](https://github.com/pouchdb/pouchdb/pull/5109) for it, but no release quite yet. I'll bug them to see when their next release is. Hopefully they publish an update soon.

In the meantime, Node 4 (LTS) and 5 will work great :)

**Comment by chrisfosterelli** [comment]
Thought: if they aren't going to have a release in the next day or so, we can look at pinning the package version to point to the Github repository until they publish. That'd fix this in the meantime.

**Comment by jonotron** [comment]
I think PouchDB 5.3.3 will fix this and should release soon. I tried depending on the github repo directly, but then we'd have to run the pouch build process which I don't really want to manage. So, we will need to wait for a new release that fixes this.

Alternatively, is there a way to coerce pouchdb to use the updated nan package?

**Comment by chrisfosterelli** [comment]
I don't think so, at least without forking the package and building/publishing it ourselves. They appear to not be using [semver ranges](https://github.com/pouchdb/pouchdb/blob/6fbec6fffa7fb40980745b7e68ca448ce6668772/package.json)

**Comment by jonotron** [comment]
Ok, let's just leave this and see if it resolves itself shortly.

**Comment by jonotron** [comment]
I've been running node `6.3` and tickbin `0.4.0`. I do not see this error. I believe this has been resolved. Can this be closed?

**Comment by chrisfosterelli** [comment]
@jonotron Yes 5.4.0 would have included the fix and we are using 5.4.5. Thanks for following up, we forgot about this!
